### PR TITLE
feat: add requested features to VM module - `avm/res/compute/virtual-machine`

### LINKED ISSUE #1760: [AVM Module Issue]: compute/virtual-machine needs support for VMSS Flex
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/compute/virtual-machine

### (Optional) Module Version

current

### Description

I want to use this module to add VMs to an existing VMSS.  There seems to be no way to actually do this, though the documentation refers multiple times to things that are not compatible with VMSS such as explicit zone assignments and putting the VM into an availability set.

The ARM reference for [Microsoft.Compute/virtualMachines](https://learn.microsoft.com/en-us/azure/templates/microsoft.compute/virtualmachines) shows that all that's needed is the resource ID of the target VMSS so hopefully this is an easy add.  I may take a shot at it myself.

  virtualMachineScaleSet: {
    id: 'string'
  }

### (Optional) Correlation Id

_No response_

### LINKED ISSUE #1775: [AVM Module Issue]: VM can't have a DNS name set on their PIP
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/compute/virtual-machine

### (Optional) Module Version

_No response_

### Description

 dnsSettings: '<dnsSettings>' is missing from module networkInterface_publicIPAddresses 'br/public:avm/res/network/public-ip-address:0.2.1' in avm/res/compute/virtual-machine/modules/nic-configuration.bicep so can't set a DNS name for VMs.

### (Optional) Correlation Id

_No response_

### LINKED ISSUE #1779: [AVM Module Issue]: Missing "supressFailures" Parameter in Bicep
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/compute/virtual-machine

### (Optional) Module Version

_No response_

### Description

In the Bicep Template the Parameter "supressFailures" for extensions is missing.
The ARM Template (json) has this Parameter. Would be great if it can also be implemented in the Bicep Template.

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by rahalan**
Fixes/Solves:
- #1760 
- #1779 
- #1775 
successful run (only NVidia fails because of MCAPS policy): [avm.res.compute.virtual-machine](https://github.com/rahalan/bicep-registry-modules/actions/workflows/avm.res.compute.virtual-machine.yml)

### CODE REVIEW

File: avm/res/compute/virtual-machine/main.bicep [hunk 1]
```diff
@@ -97,6 +97,9 @@ param bootDiagnosticStorageAccountUri string = '.blob.${environment().suffixes.s
 @description('Optional. Resource ID of a proximity placement group.')
 param proximityPlacementGroupResourceId string = ''
 
+@description('Optional. Resource ID of a virtual machine scale set, where the VM should be added.')
+param virtualMachineScaleSetId string = ''
```
Comment (AlexanderSehr) [inline avm/res/compute/virtual-machine/main.bicep:101]:
```suggestion
param virtualMachineScaleSetResourceId string = ''
```
Note, needs to be update in references as well and requires a refresh of the generated files

File: avm/res/compute/virtual-machine/main.bicep [hunk 2]
```diff
@@ -672,6 +683,9 @@ module vm_domainJoinExtension 'extension/main.bicep' =
         ? extensionDomainJoinConfig.enableAutomaticUpgrade
         : false
       settings: extensionDomainJoinConfig.settings
+      supressFailures: contains(extensionDomainJoinConfig, 'supressFailures')
+        ? extensionDomainJoinConfig.supressFailures
+        : false
```
Comment (AlexanderSehr) [inline avm/res/compute/virtual-machine/main.bicep:688]:
```suggestion
      supressFailures: extensionDomainJoinConfig.?supressFailures ?? false
```

File: avm/res/compute/virtual-machine/main.bicep [hunk 3]
```diff
@@ -701,6 +715,9 @@ module vm_microsoftAntiMalwareExtension 'extension/main.bicep' =
         ? extensionAntiMalwareConfig.enableAutomaticUpgrade
         : false
       settings: extensionAntiMalwareConfig.settings
+      supressFailures: contains(extensionAntiMalwareConfig, 'supressFailures')
+        ? extensionAntiMalwareConfig.supressFailures
+        : false
```
Comment (AlexanderSehr) [inline avm/res/compute/virtual-machine/main.bicep:720]:
```suggestion
      supressFailures: extensionAntiMalwareConfig.?supressFailures ?? false
```

File: avm/res/compute/virtual-machine/main.bicep [hunk 4]
```diff
@@ -750,6 +767,9 @@ module vm_azureMonitorAgentExtension 'extension/main.bicep' =
           : ''
         GCS_AUTO_CONFIG: osType == 'Linux' ? true : null
       }
+      supressFailures: contains(extensionMonitoringAgentConfig, 'supressFailures')
+        ? extensionMonitoringAgentConfig.supressFailures
+        : false
```
Comment (AlexanderSehr) [inline avm/res/compute/virtual-machine/main.bicep:772]:
```suggestion
      supressFailures: extensionMonitoringAgentConfig.?supressFailures ?? false
```

File: avm/res/compute/virtual-machine/main.bicep [hunk 5]
```diff
@@ -785,6 +805,9 @@ module vm_dependencyAgentExtension 'extension/main.bicep' =
           ? extensionDependencyAgentConfig.enableAMA
           : true
       }
+      supressFailures: contains(extensionDependencyAgentConfig, 'supressFailures')
+        ? extensionDependencyAgentConfig.supressFailures
+        : false
```
Comment (AlexanderSehr) [inline avm/res/compute/virtual-machine/main.bicep:810]:
```suggestion
      supressFailures: extensionDependencyAgentConfig.?supressFailures ?? false
```

File: avm/res/compute/virtual-machine/main.bicep [hunk 6]
```diff
@@ -810,6 +833,9 @@ module vm_networkWatcherAgentExtension 'extension/main.bicep' =
       enableAutomaticUpgrade: contains(extensionNetworkWatcherAgentConfig, 'enableAutomaticUpgrade')
         ? extensionNetworkWatcherAgentConfig.enableAutomaticUpgrade
         : false
+      supressFailures: contains(extensionNetworkWatcherAgentConfig, 'supressFailures')
+        ? extensionNetworkWatcherAgentConfig.supressFailures
+        : false
```
Comment (AlexanderSehr) [inline avm/res/compute/virtual-machine/main.bicep:838]:
```suggestion
      supressFailures: extensionNetworkWatcherAgentConfig.?supressFailures ?? false
```

File: avm/res/compute/virtual-machine/main.bicep [hunk 7]
```diff
@@ -836,6 +862,7 @@ module vm_desiredStateConfigurationExtension 'extension/main.bicep' =
         ? extensionDSCConfig.enableAutomaticUpgrade
         : false
       settings: contains(extensionDSCConfig, 'settings') ? extensionDSCConfig.settings : {}
+      supressFailures: contains(extensionDSCConfig, 'supressFailures') ? extensionDSCConfig.supressFailures : false
```
Comment (AlexanderSehr) [inline avm/res/compute/virtual-machine/main.bicep:865]:
```suggestion
      supressFailures: extensionDSCConfig.?supressFailures ?? false
```

File: avm/res/compute/virtual-machine/main.bicep [hunk 8]
```diff
@@ -898,7 +928,12 @@ module vm_azureDiskEncryptionExtension 'extension/main.bicep' =
       forceUpdateTag: contains(extensionAzureDiskEncryptionConfig, 'forceUpdateTag')
         ? extensionAzureDiskEncryptionConfig.forceUpdateTag
         : '1.0'
-      settings: extensionAzureDiskEncryptionConfig.settings
+      settings: contains(extensionAzureDiskEncryptionConfig, 'settings')
+        ? extensionAzureDiskEncryptionConfig.settings
+        : false
+      supressFailures: contains(extensionAzureDiskEncryptionConfig, 'supressFailures')
+        ? extensionAzureDiskEncryptionConfig.supressFailures
+        : false
```
Comment (AlexanderSehr) [inline avm/res/compute/virtual-machine/main.bicep:936]:
```suggestion
      settings: extensionAzureDiskEncryptionConfig.?settings ?? false
      supressFailures: extensionAzureDiskEncryptionConfig.?supressFailures ?? false
```

Comment (AlexanderSehr) [reply avm/res/compute/virtual-machine/main.bicep:936]: Isn't settings an object?

File: avm/res/compute/virtual-machine/main.bicep [hunk 9]
```diff
@@ -924,6 +959,9 @@ module vm_nvidiaGpuDriverWindowsExtension 'extension/main.bicep' =
       enableAutomaticUpgrade: contains(extensionNvidiaGpuDriverWindows, 'enableAutomaticUpgrade')
         ? extensionNvidiaGpuDriverWindows.enableAutomaticUpgrade
         : false
+      supressFailures: contains(extensionNvidiaGpuDriverWindows, 'supressFailures')
+        ? extensionNvidiaGpuDriverWindows.supressFailures
+        : false
```
Comment (AlexanderSehr) [inline avm/res/compute/virtual-machine/main.bicep:964]:
```suggestion
      supressFailures: extensionNvidiaGpuDriverWindows.?supressFailures ?? false
```

File: avm/res/compute/virtual-machine/main.bicep [hunk 10]
```diff
@@ -957,6 +995,9 @@ module vm_hostPoolRegistrationExtension 'extension/main.bicep' =
           registrationInfoToken: extensionHostPoolRegistration.registrationInfoToken
           aadJoin: true
         }
+        supressFailures: contains(extensionHostPoolRegistration, 'supressFailures')
+          ? extensionHostPoolRegistration.supressFailures
+          : false
```
Comment (AlexanderSehr) [inline avm/res/compute/virtual-machine/main.bicep:1000]:
```suggestion
        supressFailures: extensionHostPoolRegistration.?supressFailures ?? false
```

File: avm/res/compute/virtual-machine/main.bicep [hunk 11]
```diff
@@ -989,6 +1030,9 @@ module vm_azureGuestConfigurationExtension 'extension/main.bicep' =
       settings: contains(extensionGuestConfigurationExtension, 'settings')
         ? extensionGuestConfigurationExtension.settings
         : {}
+      supressFailures: contains(extensionGuestConfigurationExtension, 'supressFailures')
+        ? extensionGuestConfigurationExtension.supressFailures
+        : false
```
Comment (AlexanderSehr) [inline avm/res/compute/virtual-machine/main.bicep:1035]:
```suggestion
      supressFailures: extensionGuestConfigurationExtension.?supressFailures ?? false
```

File: avm/res/compute/virtual-machine/modules/nic-configuration.bicep [hunk 12]
```diff
@@ -35,6 +35,15 @@ module networkInterface_publicIPAddresses 'br/public:avm/res/network/public-ip-a
...
       lock: lock
+      idleTimeoutInMinutes: contains(ipConfiguration.pipConfiguration, 'idleTimeoutInMinutes')
+        ? ipConfiguration.pipConfiguration.idleTimeoutInMinutes
+        : null
+      ddosSettings: contains(ipConfiguration.pipConfiguration, 'ddosSettings')
+        ? ipConfiguration.pipConfiguration.ddosSettings
+        : null
+      dnsSettings: contains(ipConfiguration.pipConfiguration, 'dnsSettings')
+        ? ipConfiguration.pipConfiguration.dnsSettings
+        : null
```
Comment (AlexanderSehr) [inline avm/res/compute/virtual-machine/modules/nic-configuration.bicep:46]:
```suggestion
      idleTimeoutInMinutes: ipConfiguration.pipConfiguration.?idleTimeoutInMinutes
      ddosSettings: ipConfiguration.pipConfiguration.?ddosSettings
      dnsSettings: ipConfiguration.pipConfiguration.?dnsSettings
```

File: avm/res/compute/virtual-machine/main.bicep (hunk 8 above)
Comment (rahalan) [reply avm/res/compute/virtual-machine/main.bicep]: @AlexanderSehr well spotted!

File: avm/res/compute/virtual-machine/main.bicep (hunk 1 above)
Comment (rahalan) [reply avm/res/compute/virtual-machine/main.bicep]: @AlexanderSehr done

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**